source/mqtt_client_connection.c

```diff
@@ -106,34 +465,56 @@ napi_value mqtt_client_connection_new(napi_env env, napi_callback_info info) {
 
     node_connection->env = env;
 
-    if (node_connection->on_connection_interrupted || node_connection->on_connection_resumed) {
-        /* #TODO */
+    if (node_connection->on_connection_interupted || node_connection->on_connection_resumed) {
         aws_mqtt_client_connection_set_connection_interruption_handlers(
-            node_connection->connection, NULL, NULL, NULL, NULL);
+            node_connection->connection,
+            s_on_connection_interupted,
+            node_connection,
+            s_on_connection_resumed,
+            node_connection);
     }
 
     napi_value node_external;
-    if (napi_create_external(env, node_connection, s_node_connection_finalize, NULL, &node_external)) {
+    if (napi_create_external(env, node_connection, NULL, NULL, &node_external)) {
```

Comment:
No finalizer. So does this mean we're forcing people to ```close()``` their connections manually always?

Reply:
yes, the finalizer is worthless.